File: TeeNew/Builder2010/TeeCommander_Demo.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "TeeCommander_Demo.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "Bar3D"
#pragma link "TeeComma"
#pragma resource "*.dfm"
TTeeCommanderDemo *TeeCommanderDemo;
//---------------------------------------------------------------------------
__fastcall TTeeCommanderDemo::TTeeCommanderDemo(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TTeeCommanderDemo::CheckBox3Click(TObject *Sender)
{
  TeeCommander1->LabelText->Visible = CheckBox3->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TTeeCommanderDemo::CheckBox2Click(TObject *Sender)
{
  TeeCommander1->ParentShowHint = false;
  TeeCommander1->ShowHint = CheckBox2->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TTeeCommanderDemo::CheckBox1Click(TObject *Sender)
{
  TeeCommander1->Vertical = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

void __fastcall TTeeCommanderDemo::CheckBox4Click(TObject *Sender)
{
  if (CheckBox4->Checked) TeeCommander1->Panel = Chart1;
  else TeeCommander1->Panel = NULL;
}
//---------------------------------------------------------------------------

void __fastcall TTeeCommanderDemo::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(5);
}
//---------------------------------------------------------------------------

File: TeeNew/TeeCommander_Demo.pas
unit TeeCommander_Demo;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeeComma, TeeProcs, TeEngine, Chart, Series, Bar3D;

type
  TTeeCommanderDemo = class(TBaseForm)
    CheckBox1: TCheckBox;
    TeeCommander1: TTeeCommander;
    CheckBox2: TCheckBox;
    CheckBox3: TCheckBox;
    CheckBox4: TCheckBox;
    Series1: TBar3DSeries;
    procedure CheckBox3Click(Sender: TObject);
    procedure CheckBox2Click(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
    procedure CheckBox4Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TTeeCommanderDemo.CheckBox3Click(Sender: TObject);
begin
  TeeCommander1.LabelText.Visible:=CheckBox3.Checked;
end;

procedure TTeeCommanderDemo.CheckBox2Click(Sender: TObject);
begin
  TeeCommander1.ParentShowHint:=False;
  TeeCommander1.ShowHint:=CheckBox2.Checked;
end;

procedure TTeeCommanderDemo.CheckBox1Click(Sender: TObject);
begin
  TeeCommander1.Vertical:=CheckBox1.Checked;
end;

procedure TTeeCommanderDemo.CheckBox4Click(Sender: TObject);
begin
  if CheckBox4.Checked then TeeCommander1.Panel:=Chart1
                       else TeeCommander1.Panel:=nil;
end;

procedure TTeeCommanderDemo.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(5);
end;

initialization
  RegisterClass(TTeeCommanderDemo);
end.
